pkg/cluster/resources.go

```diff
@@ -154,6 +154,34 @@ func CreateNamespace(ctx context.Context, cli client.Client, namespace string, m
 	return desiredNamespace, client.IgnoreAlreadyExists(createErr)
 }
 
+// ProcessAllNamespace execute func() on all active namespaces in the cluster with 500 as chunksize.
+func ProcessAllNamespace(ctx context.Context, cli client.Client, processFunc func(*corev1.Namespace) error) error {
+	namespaces := &corev1.NamespaceList{}
+	paginateListOption := &client.ListOptions{
+		Limit: 500,
+	}
+
+	for { // for the case we have thousands of NS in the cluster
+		if err := cli.List(ctx, namespaces, paginateListOption); err != nil {
+			return err
+		}
+		for i := range namespaces.Items {
+			ns := &namespaces.Items[i]
+			// check namespace status if not Active, then skip
+			if ns.Status.Phase != corev1.NamespaceActive {
+				continue
+			}
```

Comment:
I wonder if this check needs to be here. Perhaps we could leave it up to decide for `processFunc`? WDYT?

Reply:
offline discussed a bit, and this is removed and up to caller to decided what is needed, in this case ShouldInjectTrustedBundle